guides/micronaut-graphql-todo/java/src/main/java/example/micronaut/AuthorRepository.java

```diff
@@ -0,0 +1,22 @@
+package example.micronaut;
+
+import io.micronaut.data.jdbc.annotation.JdbcRepository;
+import io.micronaut.data.model.query.builder.sql.Dialect;
+import io.micronaut.data.repository.CrudRepository;
+
+import javax.transaction.Transactional;
+import java.util.Collection;
+import java.util.Optional;
+
+@JdbcRepository(dialect = Dialect.POSTGRES) // <1>
+public interface AuthorRepository extends CrudRepository<Author, Long> {
+
+    Optional<Author> findByUsername(String username); // <2>
+
+    Collection<Author> findByIdInList(Collection<Long> ids); // <3>
+
+    @Transactional
```

Comment:
Repositories should _not_ be transactional except in trivial apps. Transactions should be demarcated in a business logic layer, in this case in `CreateToDoDataFetcher`